relations/src/gr1cs/constraint_system.rs

```diff
@@ -304,24 +325,238 @@ impl<F: Field> ConstraintSystem<F> {
         Ok(())
     }
 
-    /// Adds a new linear combination to the constraint system.
-    /// If the linear combination is already in the map, return the
-    /// corresponding index (using bimap)
+    /// Enforce a constraint for a predicate with arity 2.
+    pub fn enforce_constraint_arity_2(
```

Comment:
We add helpers for specific arity constraints. This avoids the dyn object required by `enforce_constraints`